backend/controller/timeline/events_log.go

```diff
@@ -0,0 +1,82 @@
+package timeline
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+
+	ftlencryption "github.com/TBD54566975/ftl/backend/controller/encryption/api"
+	"github.com/TBD54566975/ftl/backend/controller/timeline/internal/sql"
+	"github.com/TBD54566975/ftl/backend/libdal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type LogEvent struct {
+	ID            int64
+	DeploymentKey model.DeploymentKey
+	RequestKey    optional.Option[model.RequestKey]
+	Time          time.Time
+	Level         int32
+	Attributes    map[string]string
+	Message       string
+	Error         optional.Option[string]
+}
+
+func (e *LogEvent) GetID() int64 { return e.ID }
+func (e *LogEvent) event()       {}
+
+type eventLogJSON struct {
+	Message    string                  `json:"message"`
+	Attributes map[string]string       `json:"attributes"`
+	Error      optional.Option[string] `json:"error,omitempty"`
+}
+
+type Log struct {
```

Comment:
Same thing here, and probably with all the events potentially? I'd embed the struct.